File: EMS_BACKEND1/src/main/java/com/ems/service/TaskService.java
package com.ems.service;

import com.ems.dto.TaskDTO;

import java.util.List;

public interface TaskService {
    TaskDTO createTask(TaskDTO taskDTO);
    TaskDTO getTaskById(Long taskId);
    List<TaskDTO> getAllTasks();
    List<TaskDTO> getTasksByEmployeeId(Long employeeId);
    List<TaskDTO> getTasksByStatus(String status);
    List<TaskDTO> getTasksByPriority(String priority);
    TaskDTO updateTask(Long taskId, TaskDTO taskDTO);
    void deleteTask(Long taskId);
}

File: employee-management-frontend/src/services/taskService.js
// src/services/taskService.js
import axiosInstance from "../api/axiosConfig";

const TASK_API_URL = "/tasks";

const TaskService = {
  createTask: (taskData) => {
    return axiosInstance.post(TASK_API_URL, taskData);
  },

  getTaskById: (id) => {
    return axiosInstance.get(`${TASK_API_URL}/${id}`);
  },

  getAllTasks: () => {
    return axiosInstance.get(TASK_API_URL);
  },

  getTasksByEmployeeId: (employeeId) => {
    return axiosInstance.get(`${TASK_API_URL}/employee/${employeeId}`);
  },

  getTasksByStatus: (status) => {
    return axiosInstance.get(`${TASK_API_URL}/status/${status}`);
  },

  getTasksByPriority: (priority) => {
    return axiosInstance.get(`${TASK_API_URL}/priority/${priority}`);
  },

  updateTask: (id, taskData) => {
    return axiosInstance.put(`${TASK_API_URL}/${id}`, taskData);
  },

  deleteTask: (id) => {
    return axiosInstance.delete(`${TASK_API_URL}/${id}`);
  },
};

export default TaskService;
